portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,58 +29,46 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Date;
-
-
- 
+import com.google.gson.Gson; 
 
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
 	private List<Comment> comments;
 
- 	@Override
-    public void init() {
-        comments = new ArrayList<>();
-    }
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-        Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
+        String commentNumStr = getParameter(request, "number", "5");
+        int commentNum = 0;
+
+        try {
+            commentNum = Integer.parseInt(commentNumStr);
+        } catch (NumberFormatException e) {
+            System.err.println("Could not convert to int: " + commentNumStr);
+        }
 
+        Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
 
+        comments = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
             String author = (String) entity.getProperty("author");
             String text = (String) entity.getProperty("text");
             Date date = (Date) entity.getProperty("date");
```

Comment:
This is on the edge of my Java knowledge, but should these technically be in a `try` / `catch` due to possible `ClassCastException`?